app/src/main/java/com/poupa/vinylmusicplayer/discog/Discography.java

```diff
@@ -280,12 +282,20 @@ int syncWithMediaStore(Consumer<Integer> progressUpdater) {
         // Zombies are tracks that are removed but still indexed by MediaStore
         Predicate<Song> isZombie = (s) -> !(new File(s.data)).exists();
 
-        // Blacklist
+        // Blacklist and Whitelist
         final ArrayList<String> blackListedPaths = BlacklistStore.getInstance(context).getPaths();
+        final File startDirectory = PreferenceUtil.getInstance().getStartDirectory();
+        final String startPath = FileUtil.safeGetCanonicalPath(startDirectory);
         Predicate<Song> isBlackListed = (s) -> {
             for (String path : blackListedPaths) {
                 if (s.data.startsWith(path)) return true;
             }
+
+            // Whitelist
```

Comment:
Please either rename the `isBlacklisted` predicate, or create a new one (ie. `isIncluded = isWhitelisted && !isBlacklisted`) to express clearly the intention.